src/p5xr/core/p5xr.js

```diff
@@ -125,17 +144,135 @@ export default class p5xr {
     // WebXR availabilty
     if (navigator?.xr) {
       console.log('XR Available');
-      const mode = this.isVR ? 'VR' : 'AR';
-      const session = this.isVR ? 'immersive-vr' : 'immersive-ar';
-      const supported = await navigator.xr.isSessionSupported(session);
+      const supported = await navigator.xr.isSessionSupported(this.mode);
       this.hasImmersive = supported;
-      this.xrButton.setAvailable(supported, mode);
+      this.xrButton.setAvailable(supported, this.mode);
     } else {
       console.log('XR Not Available');
       this.xrButton.disable();
     }
   }
 
+  /**
+   * Helper function to reset XR and GL, should be called between
+   * ending an XR session and starting a new XR session
+   * @method resetXR
+   */
+  resetXR() {
+    this.xrDevice = null;
+    this.xrSession = null;
+    this.xrRefSpace = null;
+    this.xrViewerSpace = null;
+    this.xrHitTestSource = null;
+    this.gl = null;
+    this.frame = null;
+  }
+
+  /**
+   * `navigator.xr.requestSession()` must be called within a user gesture event.
+   * @private
+   * @ignore
+   */
+  __onXRButtonClicked() {
+    if (this.hasImmersive) {
+      console.log(`Requesting session with mode: ${this.mode}`);
+      this.isImmersive = true;
+      this.resetXR();
+      navigator.xr
+        .requestSession(this.mode, {
+          requiredFeatures: this.requiredFeatures,
+          optionalFeatures: this.optionalFeatures,
+        })
+        .then(this.__startSketch.bind(this))
+        .catch((error) => {
+          console.error(`An error occured activating ${this.mode}: ${error}`);
+        });
+    } else {
+      this.xrButton.hide();
+    }
+  }
+
+  /**
+   * This is where the actual p5 canvas is first created, and
+   * the GL rendering context is accessed by p5vr.
+   * The current XRSession also gets a frame of reference and
+   * base rendering layer. <br>
+   * @param {XRSession}
+   * @private
+   * @ignore
+   */
+  __startSketch(session) {
+    this.xrSession = session;
+    this.canvas = p5.instance.canvas;
+    this.canvas.style.visibility = 'visible';
+
+    if (typeof window.setup === 'function') {
+      window.setup();
+      p5.instance._millisStart = window.performance.now();
+    }
+    // const refSpaceRequest = this.isImmersive ? 'local' : 'viewer';
+    // this.xrSession.requestReferenceSpace(refSpaceRequest).then((refSpace) => {
+    //   this.xrRefSpace = refSpace;
+    //   // Inform the session that we're ready to begin drawing.
+    //   this.xrSession.requestAnimationFrame(this.__onXRFrame.bind(this));
+    //   if (!this.isImmersive) {
+    //     this.xrSession.updateRenderState({
+    //       baseLayer: new XRWebGLLayer(this.xrSession, this.gl),
+    //       inlineVerticalFieldOfView: 70 * (Math.PI / 180),
+    //     });
+    //     this.addInlineViewListeners(this.canvas);
+    //   }
+    // });
```

Comment:
I am a fan of either commenting to explain a commented out chunk or just deleting the deprecated code

Reply:
I'm putting this back, this is necessary for inline mode to work